### PR TITLE
Support update rusage in wait4 syscall

### PR DESCRIPTION
**Pull request by cchanging**
Based on PR #733.

This PR add the missing parameter `rusage` to `wait4` syscall. This parameter is used to record some statistics about the waited process. 

Currently `rusage` only records cpu time. With this PR and additionally adding a dummy `vfork` using `fork`, we can get the correct results from `time` command.

This PR also add a `resource` module which is aimed at providing structs and APIs for resources managements.

### CODE REVIEW

**Comment by tatetian** [comment]
@StevenJiang1110 Could you take a look at this PR?

File: kernel/aster-nix/src/resource/mod.rs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Resource control and accounting module.
+pub mod c_type;
```
Comment (StevenJiang1110) [inline kernel/aster-nix/src/resource/mod.rs:4]: Does `rusage` will be used by other syscalls?

If so, add `resource` as a module under `aster-nix/util`, they are commonly-used structures among maybe many syscalls.

If not, just define `rusage` in `wait4.rs`.

**Comment by tatetian** [comment]
@cchanging Need to rebase this PR

**Comment by cchanging** [comment]
Modification: Add the syscall `getrusage` which will be used in `lmbench` and update the Linux Compatibility section of the Asterinas Book.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**